client/homescreen/index.tsx

```diff
@@ -8,37 +8,46 @@ import { getSetting } from '@woocommerce/wc-admin-settings';
 import {
 	ONBOARDING_STORE_NAME,
 	withOnboardingHydration,
+	WCDataSelector,
 } from '@woocommerce/data';
 import { getHistory, getNewPath } from '@woocommerce/navigation';
-
+import type { History } from 'history';
 /**
  * Internal dependencies
  */
 import Layout from './layout';
 
-const Homescreen = ( { profileItems, query } ) => {
+type HomescreenProps = ReturnType< typeof withSelectHandler > & {
```

Comment:
This seems to work pretty well to type the compose props. for other components, I'd make this a composition of withSelect and withDispatch.